Gems/AtomLyIntegration/CommonFeatures/Code/Source/PostProcess/EditorModeFeedback/EditorEditorModeFeedbackSystemComponent.cpp

```diff
@@ -0,0 +1,268 @@
+#include <AzToolsFramework/Viewport/ViewportMessages.h>
+#include <AzToolsFramework/FocusMode/FocusModeInterface.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <Atom/RPI.Public/MeshDrawPacket.h>
+#include <Atom/RPI.Public/Shader/ShaderResourceGroup.h>
+#include <Atom/RPI.Public/Model/ModelLodUtils.h>
+#include <Atom/RPI.Public/MeshDrawPacket.h>
+#include <Atom/RPI.Public/Model/Model.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/ViewportContext.h>
+#include <Atom/RPI.Public/ViewportContextBus.h>
+#include <Atom/RPI.Public/DynamicDraw/DynamicDrawInterface.h>
+#include <Atom/Utils/Utils.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        //! Builds the mesh draw packets for the specified model.
+        static AZStd::vector<RPI::MeshDrawPacket> BuildMeshDrawPackets(
+            const RPI::View* view,
+            const AZ::Transform& worldTM,
+            const Data::Asset<RPI::ModelAsset>& modelAsset,
+            const Data::Instance<RPI::Material>& material,
+            const Data::Instance<RPI::ShaderResourceGroup>& meshObjectSrg)
+        {
+            AZStd::vector<RPI::MeshDrawPacket> meshDrawPackets;
+            const RPI::Model& model = *RPI::Model::FindOrCreate(modelAsset);
+            const auto modelLodIndex = RPI::ModelLodUtils::SelectLod(view, worldTM, model);
+            const Data::Asset<RPI::ModelLodAsset>& modelLodAsset = modelAsset->GetLodAssets()[0];
+            RPI::ModelLod& modelLod = *RPI::ModelLod::FindOrCreate(modelLodAsset, modelAsset).get();
+        
+            for (auto i = 0; i < modelLod.GetMeshes().size(); i++)
+            {
+                RPI::MeshDrawPacket drawPacket(modelLod, i, material, meshObjectSrg);
+                meshDrawPackets.push_back(AZStd::move(drawPacket));
+            }
+        
+            return meshDrawPackets;
+        }
+
+        //! Creates the mask shader resource group for the specified mesh.
+        static Data::Instance<RPI::ShaderResourceGroup> CreateMaskShaderResourceGroup(
+            const MeshFeatureProcessorInterface* featureProcessor,
+            const MeshFeatureProcessorInterface::MeshHandle& meshHandle,
+            Data::Instance<RPI::Material> maskMaterial)
+        {
+            const auto& shaderAsset = maskMaterial->GetAsset()->GetMaterialTypeAsset()->GetShaderAssetForObjectSrg();
+            const auto& objectSrgLayout = maskMaterial->GetAsset()->GetObjectSrgLayout();
+            const auto maskMeshObjectSrg = RPI::ShaderResourceGroup::Create(shaderAsset, objectSrgLayout->GetName());
+
+            // Set the object id so the correct MVP matrices can be selected in the shader
+            const auto objectId = featureProcessor->GetObjectId(meshHandle).GetIndex();
+            RHI::ShaderInputNameIndex objectIdIndex = "m_objectId";
+            maskMeshObjectSrg->SetConstant(objectIdIndex, objectId);
+
+            //! Set the id to write to the entity mask texture.
```

Comment:
nit: avoid `//!` inline in a function body (repeated below)

Reply:
Fixed 👍